content/open-source/wolfi/building-a-wolfi-package.md

```diff
@@ -0,0 +1,437 @@
+- **License**: the package license. It is important to note that only packages with OSI-approved licenses can be included in Wolfi. You can check the relevant package info in the [licenses page at opensource.org](https://opensource.org/licenses/).
+- **Runtime dependencies**: any dependencies needed by your package at runtime. Not to be confused with build dependencies, these will come up in the environment section of the file.
+
+### The `environment` Section
+
+The next section is the environment section. It defines how the build environment should look in order to build your package. Packages listed in this section won't be included in the final package, because they are only needed at build time.
+
+When building locally, you'll also need to include information about where to find Wolfi packages. This is not needed when submitting the package to the Wolfi OS repository. The `contents` node is used for that:
+
+```yaml
+environment:
+  contents:
+    repositories:
+      - https://packages.wolfi.dev/bootstrap/stage3
+      - https://packages.wolfi.dev/os
+    keyring:
+      - https://packages.wolfi.dev/bootstrap/stage3/wolfi-signing.rsa.pub
+      - https://packages.wolfi.dev/os/wolfi-signing.rsa.pub
+```
+
+Dependencies are defined in a `packages` section. An example excerpt from the Wolfi PHP package, which is a fairly complex build with many dependencies, is below:
+
+```yaml
+environment:
+  contents:
+    packages:
+      - build-base
+      - busybox
+      - file
+      - bison
+      - libtool
+      - ca-certificates-bundle
+      - bzip2-dev
+      - libxml2-dev
+      - curl-dev
+      - openssl-dev
+      - readline-dev
+      - sqlite-dev
+      - libsodium-dev
+      - libpng-dev
+      - libjpeg-turbo-dev
+      - libavif-dev
+      - libwebp-dev
+      - libxpm-dev
+      - libx11-dev
+      - freetype-dev
+      - gmp-dev
+      - icu-dev
+      - openldap-dev
+      - oniguruma-dev
+      - libxslt-dev
+      - postgresql-15-dev
+      - libzip
+```
+
+Don't worry if you don't know everything you'll need upfront at build time. Even if you build the package locally first, your system most likely has many dependencies already installed; by paying attention to the output provided by melange, you will be able to figure out what is missing, and iterate until your build environment looks right.
+
+One thing that may happen during this process is finding out that one or more dependencies needed by your package are not yet available in Wolfi, so they need to be built first. It is a normal part of the process, so don't worry – you will be able to build incrementally and test everything locally.
```

Comment:
```suggestion
One thing that may happen during this process is finding out that one or more dependencies needed by your package are not yet available in Wolfi, so they need to be built first. It is a normal part of the process, so don't worry — you will be able to build incrementally and test everything locally.
```

Reply:
I realize I'm being a stickler about em dashes, but it's not a rule in the style guide. Maybe I'll get around to adding it

Reply:
oops! :) I think it's an habit, sorry haha